### DISCUSSION TITLE
How to pickup the parameter count in REXX?

### DISCUSSION THREAD

**Discussion by Peter-Jacob** [adesutherland/CREXX#219] [Q&A]
In some bifs I need to distinguish between empty and non-existing parameters: 

translate ('abcdef')      means set string to upper cases
translate ('abcdef','','') means empty translation tables, - do nothing

in RXAS it's simple, you can use a0 as the number of parms set, this is not available in REXXC, you can't use the `assembler load parms,a0 `as they are translated into appropriate registers. we need an alternative in REXXC, maybe a compiler function arg(0)?

**Comment by adesutherland** [comment]
At the moment the only way to do this with in rexx is detecting the default value

e.g.
```
translate: procedure 
arg a1 - .string, a2="!!!!!!"
```
And detect "!!!!!" - very ugly!

Option 2 is to get the register flag and detect if it is an optional value

e.g. (untested - and you are breaking new ground)
```
translate: procedure 
arg a1 - .string, a2=""
int flag = 0;

assemble gettp flag,a2 /* flag - Bit 1 - REGTP_VAL - ONLY used for optional arguments; setting (1) means the register has a specified value */

flag = flag bitwise-and 1 (how !!!) 

if flag = 0 then do
  /* do uppercasing *.

etc.
```

**Reply by Peter-Jacob** [reply]
As I always say: if you can't solve problems with force, use brute force!
